### PR TITLE
feat: Added register view

### LINKED ISSUE #6: Auth: Website (Register)
Depends: HF5-Project/api#9 HF5-Project/overnites#5

Create UI and call to the Auth Endpoint for register

### LINKED ISSUE #7: Auth: Website (Login)
Depends: HF5-Project/api#9 HF5-Project/overnites#5

Create UI and call to the Auth Endpoint for login

### PR DESCRIPTION
**Pull request by Anders164a**
## Description

Register view has been added - Which connects to /api/auth/register and creates a user, if the written data fields are correct.

Base button, label, input & register-form has been created, to display the form with our standard colors, spacing, fonts etc.

Generel theme colors and fontFamily has been set in tailwind config.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/components/registerForm.vue [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min" style="background-color: #313030;">
+        <div class="p-5">
+            <h1>Register</h1>
+            <p>Fill all fields</p>
+            <div class="mt-12">
+                <form>
+                    <p v-if="parsedMessage" class="text-red-600 w-full mb-4 text-xs">{{ parsedMessage }}</p>
+                    <div class="input-div">
+                        <input class="w-full darkcolor lightbg mb-6 p-1.5 rounded-xl" type="text" placeholder="Name" v-model="name" />
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Create a BaseInput component to use instead of input

File: src/components/registerForm.vue [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            validPassword: false,
+            passwordsMatch: false
+        }
+    },
+    props: {
+        endpoint: {
+            type: String,
+            required: true,
+        }
+    },
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Should be static, no need for a prop

File: src/pages/register/index.astro [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+import Layout from '../../layouts/Layout.astro';
+import BaseRegister from "../../components/registerForm.vue";
```
Comment (frederikpyt) [inline src/pages/register/index.astro]:
```suggestion
import registerForm from "../../components/registerForm.vue";
```

File: src/pages/register/index.astro [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+---
+
+<Layout title="Register">
+	<div class="flex h-full content-center place-items-center">
+		<BaseRegister client:load endpoint="/api/auth/register"></BaseRegister>
```
Comment (frederikpyt) [inline src/pages/register/index.astro]:
```suggestion
		<registerForm client:load></registerForm>
```

File: tailwind.config.cjs [hunk 5]
```diff
@@ -3,6 +3,9 @@ module.exports = {
 	content: ['./src/**/*.{astro,html,js,jsx,md,mdx,svelte,ts,tsx,vue}'],
 	theme: {
 		extend: {},
+		container: {
+			center: true,
+		}
```
Comment (frederikpyt) [inline tailwind.config.cjs]: Remove

File: src/layouts/Layout.astro [hunk 6]
```diff
@@ -20,16 +20,22 @@ const { title } = Astro.props;
...
 	html {
 		font-family: system-ui, sans-serif;
-		background-color: #F6F6F6;
+		background-color: #212121;
+		color: #cfb39d;
+		height: 100%;
+	}
+	body {
+		height: 100%;
 	}
```
Comment (frederikpyt) [inline src/layouts/Layout.astro]: Can be replaced with `<body class="h-full">...</body>`

File: src/layouts/Layout.astro [hunk 7]
```diff
@@ -20,16 +20,22 @@ const { title } = Astro.props;
...
-	:root {
-		--accent: 124, 58, 237;
-		--accent-gradient: linear-gradient(45deg, rgb(var(--accent)), #da62c4 30%, white 60%);
-	}
 	html {
 		font-family: system-ui, sans-serif;
-		background-color: #F6F6F6;
+		background-color: #212121;
+		color: #cfb39d;
+		height: 100%;
```
Comment (frederikpyt) [inline src/layouts/Layout.astro]: Should be replaced with `<html class="min-h-screen">...</html>`

It is equivalent with min-height: 100vh;

File: src/components/registerForm.vue [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    }
+</script>
+<style lang="scss">
+	.input-div {
+		width: 300px;
+	}
+	input::placeholder {
+		color: #56483E;
+		font-weight: 500;
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Can be replaced with `<input class="placeholder:font-medium">`

File: src/components/registerForm.vue [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    },
+    watch: {
+        password() {
+            const length = this.password.length > 8;
+            const containsUppercase = /[A-Z]/.test(this.password);
+            const containsLowercase = /[a-z]/.test(this.password);
+            const containsNumber = /[0-9]/.test(this.password);
+            const containsSpecial = /[#?!@$%^&*-]/.test(this.password);
+            this.validPassword = length && containsUppercase && containsLowercase && containsNumber && containsSpecial;
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Should be handled by backend, just remove and we will fix later

File: src/components/registerForm.vue [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        },
+        confirmPassword() {
+            this.passwordsMatch = this.password === this.confirmPassword;
+        }
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Make it a computed function instead

File: src/components/registerForm.vue [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+                    "password": this.password || null
+                })
+            })
+            .then(async (response) => {
+                let json = await response.json();
+                if (json.statusCode !== undefined && json.statusCode !== 201) {
+                    this.errorMessage = json.message;
+                } else {
+                    window.location.href = "/";
+                }
```
Comment (frederikpyt) [inline src/components/registerForm.vue]:
```suggestion
                const json = await response.json();
                if (json.statusCode === 201) {
                    window.location.href = "/";
                } else {
                    this.errorMessage = json.message;
                }
```

File: src/components/registerForm.vue [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+        endpoint: {
+            type: String,
+            required: true,
+        }
+    },
+    computed: {
+        parsedMessage(): String {
+            return this.errorMessage;
+        }
+    },
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Unnessesary, should just call errorMessage instead.

File: src/components/registerForm.vue [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    watch: {
+        password() {
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Should also be a computed function

File: src/layouts/Layout.astro [hunk 14]
```diff
@@ -20,16 +20,22 @@ const { title } = Astro.props;
...
 <style is:global>
-	:root {
-		--accent: 124, 58, 237;
-		--accent-gradient: linear-gradient(45deg, rgb(var(--accent)), #da62c4 30%, white 60%);
-	}
 	html {
 		font-family: system-ui, sans-serif;
-		background-color: #F6F6F6;
+		background-color: #212121;
+		color: #cfb39d;
```
Comment (frederikpyt) [inline src/layouts/Layout.astro]: Take advantage of tailwindcss's color system

See: https://tailwindcss.com/docs/customizing-colors

**Review by frederikpyt — CHANGES_REQUESTED** [review]
You also forgot to write a description of changes in your pull request

File: src/layouts/Layout.astro (hunk 14 above)
Comment (kristianbinau) [reply src/layouts/Layout.astro]: https://tailwindcss.com/docs/theme this link is better

File: src/pages/register/index.astro (hunk 4 above)
Comment (Anders164a) [reply src/pages/register/index.astro]: Changes added

File: tailwind.config.cjs (hunk 5 above)
Comment (Anders164a) [reply tailwind.config.cjs]: Changes added

File: src/components/registerForm.vue [hunk 15]
```diff
@@ -0,0 +1,124 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min" style="background-color: #313030;">
+        <div class="p-5">
+            <h1>Register</h1>
+            <p>Fill all fields</p>
+            <div class="mt-12">
+                <form>
+                    <p v-if="parsedMessage" class="text-red-600 w-full mb-4 text-xs">{{ parsedMessage }}</p>
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Should be placed on the submit button instead

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Changes

File: src/layouts/Layout.astro (hunk 6 above)
Comment (Anders164a) [reply src/layouts/Layout.astro]: Changes added

File: src/layouts/Layout.astro (hunk 7 above)
Comment (Anders164a) [reply src/layouts/Layout.astro]: Changes added

File: src/components/registerForm.vue (hunk 8 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/pages/register/index.astro (hunk 3 above)
Comment (Anders164a) [reply src/pages/register/index.astro]: Changes added

File: src/components/registerForm.vue (hunk 2 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue [hunk 16]
```diff
@@ -0,0 +1,124 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min" style="background-color: #313030;">
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Should have same dimensions as design. Overall it should be more inline with design

File: src/components/registerForm.vue (hunk 11 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 9 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 1 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/layouts/Layout.astro (hunk 14 above)
Comment (Anders164a) [reply src/layouts/Layout.astro]: Changes added

File: src/components/registerForm.vue (hunk 12 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Removed

File: src/components/registerForm.vue (hunk 10 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 13 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 15 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue [hunk 17]
```diff
@@ -0,0 +1,93 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min bg-primary-600">
+        <div class="p-5">
+            <h1>Register</h1>
+            <p>Fill all fields</p>
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Whole page uses wrong font

File: src/components/registerForm.vue [hunk 18]
```diff
@@ -0,0 +1,98 @@
+<template>
+    <div class="mx-auto p-12 rounded-[50px] w-min bg-primary-600">
+        <div class="p-5">
+            <h1 class="text-3xl font-medium mb-1">Register</h1>
+            <p>Fill all fields</p>
+            <div class="mt-12">
+                <form class="space-y-5">
+                    <div class="input-div">
+                        <BaseInput id="name" type="text" v-model="name" :error-text="nameErrorMessage" place-holder="Name"></BaseInput>
```
Comment (frederikpyt) [inline src/components/registerForm.vue]:
```suggestion
                        <BaseInput id="name" type="text" v-model="name" :error-text="nameErrorMessage" place-holder="Name" auto-focus="true"></BaseInput>
```

File: src/components/registerForm.vue [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+                this.errorMessage = response.message;
+            })
+        }
+    }
+    }
+</script>
+<style lang="scss">
+    .input-div {
+        width: 350px;
+    }
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Try to do this with tailwindcss

File: src/components/registerForm.vue [hunk 20]
```diff
@@ -0,0 +1,93 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min bg-primary-600">
+        <div class="p-5">
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Too much spacing top and bottom

File: src/components/registerForm.vue [hunk 21]
```diff
@@ -0,0 +1,98 @@
+<template>
+    <div class="mx-auto p-12 rounded-[50px] w-min bg-primary-600">
+        <div class="p-5">
+            <h1 class="text-3xl font-medium mb-1">Register</h1>
+            <p>Fill all fields</p>
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Setup i18n for multi language support

**Review by frederikpyt — CHANGES_REQUESTED** [review]

**Comment by frederikpyt** [comment]
The name of the PR should be `feat: Added register view`

You are also missing this from the previous review:
> You also forgot to write a description of changes in your pull request

File: src/components/registerForm.vue (hunk 16 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 18 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+                    </div>
+                    <div class="input-div">
+                        <BaseInput id="passwordConfirm" type="password" v-model="confirmPassword" :error-text="passwordConfirmErrorMessage" place-holder="Confirm Password"></BaseInput>
+                    </div>
+                    <div>
+                        <span v-if="errorMessage" class="text-red-600 w-full ml-2 mb-4 text-xs">{{ errorMessage }}</span>
+                        <BaseButton type="submit" @click="register">Register</BaseButton>
+                    </div>
+                    <div>
+                        <p class="text-primary-300 font-bold w-full mb-4 text-xs">Already have an account? Click <a class="text-primary-200 underline" href="/login">here</a>, to log in.</p>
```
Comment (kristianbinau) [inline src/components/registerForm.vue]: Wrong font-size and weight

File: src/components/registerForm.vue (hunk 21 above)
Comment (frederikpyt) [reply src/components/registerForm.vue]: An issue has been made for this https://github.com/HF5-PROJECT/web/issues/12

Comment (kristianbinau) [inline src/components/registerForm.vue]: Wrong font size

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Fonts are wrong quite a few places. Check them all : )

File: src/components/registerForm.vue (hunk 19 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 17 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Font added

File: src/components/registerForm.vue (hunk 22 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 21 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue (hunk 20 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/registerForm.vue [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+                method: "POST",
+                headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    "name": this.name || null,
+                    "email": this.email || null,
+                    "address": "",
+                    "password": this.password || null
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Why the or null?

File: src/components/registerForm.vue [hunk 24]
```diff
@@ -0,0 +1,93 @@
+<template>
+    <div class="mx-auto p-12 rounded-[50px] w-min bg-primary-600">
+        <div class="p-4">
+            <h1 class="text-3xl font-medium mb-1">Register</h1>
+            <p>Fill all fields</p>
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Use the new i18n implementation

**Review by frederikpyt — CHANGES_REQUESTED** [review]

**Review by kristianbinau — APPROVED** [review]
I'll let @frederikpyt handle the rest. Good job! 🍭

File: src/components/registerForm.vue (hunk 23 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Null and address removed

File: src/components/registerForm.vue (hunk 24 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/loginForm.vue [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+    BaseInput,
+    BaseButton,
+  },
+  data() {
+    return {
+      email: "",
+      password: "",
+      errorMessage: "",
+      emailErrorMessage: "",
+      passwordErrorMessage: "",
```
Comment (frederikpyt) [inline src/components/loginForm.vue]: Remove

File: src/components/loginForm.vue [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+      <h1 class="text-3xl font-medium mb-1">{{ $t("auth.login") }}</h1>
+      <p>{{ $t("auth.fillOutYourInfo") }}</p>
+      <div class="mt-12">
+        <form class="space-y-5">
+          <div class="w-96">
+            <BaseInput
+              id="email"
+              type="email"
+              v-model="email"
+              :error-text="emailErrorMessage"
```
Comment (frederikpyt) [inline src/components/loginForm.vue]: Remove

File: src/components/loginForm.vue [hunk 27]
```diff
@@ -0,0 +1,92 @@
...
+              :place-holder="$t('auth.email')"
+            ></BaseInput>
+          </div>
+          <div class="w-96">
+            <BaseInput
+              id="password"
+              type="password"
+              v-model="password"
+              :error-text="passwordErrorMessage"
```
Comment (frederikpyt) [inline src/components/loginForm.vue]: Remove

File: src/components/registerForm.vue [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+    },
+    computed: {
+        doesPasswordMatch() {
+            return this.password === this.confirmPassword;
+        }
+    },
+    methods: {
+        register(): void {
+            if (!this.doesPasswordMatch) {
+                this.errorMessage = this.$t("auth.passwordsIdentical");
```
Comment (frederikpyt) [inline src/components/registerForm.vue]: Maybe change this to another errormessage, perhaps the passwordErrorMessage and/or passwordConfirmErrorMessage

**Review by frederikpyt — CHANGES_REQUESTED** [review]

File: src/components/loginForm.vue (hunk 25 above)
Comment (Anders164a) [reply src/components/loginForm.vue]: Removed

File: src/components/loginForm.vue (hunk 26 above)
Comment (Anders164a) [reply src/components/loginForm.vue]: Removed

File: src/components/loginForm.vue (hunk 27 above)
Comment (Anders164a) [reply src/components/loginForm.vue]: Removed

File: src/components/registerForm.vue (hunk 28 above)
Comment (Anders164a) [reply src/components/registerForm.vue]: Changes added

File: src/components/loginForm.vue [hunk 29]
```diff
@@ -62,6 +62,7 @@ export default {
   },
```
Comment (frederikpyt) [inline src/components/loginForm.vue:58]: Remove emailErrorMessage and passwordErrorMessage

**Review by frederikpyt — CHANGES_REQUESTED** [review]

File: src/components/loginForm.vue (hunk 29 above)
Comment (Anders164a) [reply src/components/loginForm.vue:58]: Changes added

**Review by frederikpyt — APPROVED** [review]

**Pull request merged**